apps/website/src/composables/wallet.ts

```diff
@@ -1,47 +1,69 @@
 import { ref } from 'vue'
 import { ethers } from 'ethers'
 import { WalletProvider } from '@/interfaces/WalletProvider'
+import useIopay from '@/composables/iopay'
+
+const { getIoPayAccounts, getIoPayProvider, sendIoPayTransaction } = useIopay()
 
 const defaultProviders = {
   metamask: undefined,
   coinbase: undefined,
+  iopay: undefined,
 }
 
 export default function useWallet() {
   const ethereum: any = window.ethereum
   const availableProviders = ref<Record<string, WalletProvider>>(
     getAvailableProviders(ethereum)
   )
-  const selectedProvider = ref<WalletProvider>({})
+  const selectedProvider = ref<WalletProvider | string>({}) // Adding string type to support 'iopay' throws new ts errors in sendTransaction
   const selectedAccount = ref<string>('')
-  const toAddress = ref<string>('') // Test to address: 0xD4e5faa8aD7d499Aa03BDDE2a3116E66bc8F8203
-  const amount = ref<string>('')
+  const amount = ref<string>('') // Use '1' to test
+  const toAddress = ref<string>('')
+  // Test ethereum send to address : 0xD4e5faa8aD7d499Aa03BDDE2a3116E66bc8F8203
+  // Test iotex send to address: acc://06da5e904240736b1e21ca6dbbd5f619860803af04ff3d54/acme
 
   async function connectWallet(provider: string) {
     try {
-      selectedProvider.value = availableProviders.value[provider]
-      if (!selectedProvider.value) {
-        throw new Error('No provider selected')
+      if (provider === 'iopay') {
+        const accounts = await getIoPayAccounts()
+        const { address } = accounts[0]
+        selectedProvider.value = getIoPayProvider()
+        selectedAccount.value = address
+      } else {
+        selectedProvider.value = availableProviders.value[provider]
+        if (!selectedProvider.value) {
+          throw new Error('No provider selected')
+        }
+        selectedAccount.value = await requestAccount(selectedProvider.value)
       }
-      selectedAccount.value = await requestAccount(selectedProvider.value)
     } catch (error) {
       console.error(error)
     }
   }
 
   async function sendTransaction() {
     try {
-      const web3Provider: ethers.providers.Web3Provider =
-        new ethers.providers.Web3Provider(selectedProvider.value)
-      const signer = web3Provider.getSigner()
-      const etherAmount = ethers.utils.parseEther(amount.value)
-      const tx = {
-        to: toAddress.value,
-        value: etherAmount,
+      if (
+        // TODO: Understand and fix typescript error on next 5 lines
```

Comment:
Switching to string ref for `selectedProvider` will get rid of the errors.

Otherwise, you would just need to assert the types before using them, like `(selectedProvider.value as WalletProvider).isMetaMask`. 

Refactor to get the provider object or call provider method by lookup string.

Reply:
[Here](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) are docs on `type assertions` – good to have but we'll avoid the extra code in this case by going with strings or string unions.